en/contribute/README.md

```diff
@@ -103,7 +103,79 @@ To contribute new functionality, [sign up for Github](https://help.github.com/ar
 
 All code contributions have to be under the permissive [BSD 3-clause license](https://opensource.org/licenses/BSD-3-Clause) and all code must not impose any further constraints on the use.
 
-### Code Style Formatting
+### Coding Standards {#code_standards}
+
+PX4 recommends the following coding standards:
+- In-source documentation [as below](#source_docs)
+- Formatting/layout using *astyle*, [as below](#code_formatting)
+- All files should have standard copyright header, with date range reflecting creation to last modification date
+  (note that the comment markup may change for different types of files).
+  ```
+  /****************************************************************************
+  *
+  *   Copyright (c) 2013-2019 PX4 Development Team. All rights reserved.
+  *
+  * Redistribution and use in source and binary forms, with or without
+  * modification, are permitted provided that the following conditions
+  * are met:
+  *
+  * 1. Redistributions of source code must retain the above copyright
+  *    notice, this list of conditions and the following disclaimer.
+  * 2. Redistributions in binary form must reproduce the above copyright
+  *    notice, this list of conditions and the following disclaimer in
+  *    the documentation and/or other materials provided with the
+  *    distribution.
+  * 3. Neither the name PX4 nor the names of its contributors may be
+  *    used to endorse or promote products derived from this software
+  *    without specific prior written permission.
+  *
+  * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+  * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+  * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+  * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+  * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+  * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+  * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+  * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+  * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+  * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+  * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+  * POSSIBILITY OF SUCH DAMAGE.
+  *
+  ****************************************************************************/
+  ```
+- Naming conventions:
+  - Classes
+    - Use CamelCase
+    - Start with a capitalized letter and are nouns.
+    - One class per file, with filename prefix matching classname.
+  - Methods/functions and variables
+    - Use snake case (words start with lower case letters and are separated by underscores).
+    - Files containing just functions may be named following pattern of main function.
+  - Member variables
+    - Start with an underscore and otherwise use snake case.
+- Prefer `#pragma once` over include guards
+
+
+<!-- 
```

Comment:
Note, I did not include the headers section below (from Mark's original post). Do we follow these rules? If this is something astyle can and should enforce, can we push it into the tool?